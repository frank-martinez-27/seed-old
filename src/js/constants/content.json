{
  "messages": {
    "en": {
      "seed.containers.AppShell.header": "Welcome",
      "seed.containers.Modal.close.button": "Close",
      "seed.containers.Modal.errorHeader": "Error Header",
      "seed.containers.Modal.close.buttonText": "Close"
    },
    "es": {
      "seed.containers.AppShell.header": "Bienvenidos",
      "seed.containers.Modal.close.button": "Cerrar",
      "seed.containers.Modal.errorHeader": "Titulo Error",
      "seed.containers.Modal.close.buttonText": "Cerrar"
    }
  }
}