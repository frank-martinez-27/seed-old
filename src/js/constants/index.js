export const ACTIONS = Object.freeze({
  SHOW_MODAL: 'SHOW_MODAL',
  HIDE_MODAL: 'HIDE_MODAL',
  AUTHORIZE_REQUEST: 'AUTHORIZE_REQUEST',
  AUTHORIZE_SUCCESS: 'AUTHORIZE_SUCCESS',
  AUTHORIZE_ERROR: 'AUTHORIZE_ERROR',
  KEEP_ALIVE_REQUEST: 'KEEP_ALIVE_REQUEST',
  KEEP_ALIVE_SUCCESS: 'KEEP_ALIVE_SUCCESS',
  KEEP_ALIVE_ERROR: 'KEEP_ALIVE_ERROR',
  LOG_ERRORS: 'LOG_ERRORS',
  RESET_ERRORS: 'RESET_ERRORS',
  LOAD_APP_CONTENT_REQUEST: 'LOAD_APP_CONTENT_REQUEST',
  LOAD_APP_CONTENT_SUCCESS: 'LOAD_APP_CONTENT_SUCCESS',
  LOAD_APP_CONTENT_ERROR: 'LOAD_APP_CONTENT_ERROR',
});

export const CONTEXT_ROOT = '/';

export const API_ROOTS = Object.freeze({
  MY_APP: '/myApp/api/v1',
  GITHUB_PAGES: 'https://github.com',
});

export const API_PATHS = Object.freeze({
  TEST: '/test',
  KEEP_ALIVE: '/keepAlive',
  AUTH_USER: '/api/1/cookie/validate',
  USER_PROFILE: '/user/profile',
  USERS: '/users',
});

export const ROUTE_PATHS = Object.freeze({
  INDEX: '/',
  PROCESS: '/myApp/:id',
  ERROR: '/error',
});

export const MODALS = Object.freeze({
  NONE: 'NONE',
  ERROR: 'ERROR',
});

export const KEYS = Object.freeze({
  DOWN: 40,
  ENTER: 13,
  ESC: 27,
  SPACE: 32,
  UP: 38,
});

export const HTTP = Object.freeze({
  STATUS: {
    OK: 200,
    MULTIPLE_CHOICES: 300,
    BAD_REQUEST: 400,
    UNAUTHORIZED: 401,
    FORBIDDEN: 403,
    NOT_FOUND: 404,
    CONFLICT: 409,
    INTERNAL_SERVER_ERROR: 500,
  },
  REQUEST_METHOD: {
    GET: 'GET',
    POST: 'POST',
    PUT: 'PUT',
    DELETE: 'DELETE',
  },
  HEADER_FIELD: {
    CONTENT_TYPE: 'Content-Type',
    AUTHORIZATION: 'Authorization',
    X_SATI: 'X-SATI',
  },
  MIME_TYPE: {
    JSON: 'application/json',
  },
  CREDENTIALS: 'credentials',
  SAME_ORIGIN: 'same-origin',
});

export const LANGUAGES = Object.freeze({
  ENGLISH: 'en',
  ESPANOL: 'es',
});

export const KEEP_ALIVE = Object.freeze({
  EVENTS: ['mousemove', 'mousedown', 'keydown', 'touchstart', 'scroll'],
  DELAY: 10000,
  OPTIONS: { maxWait: 30000 },
  IDLE_TIMEOUT: 2100000, // 35 minutes
});

export const ERRORS = Object.freeze({
  ERROR: 'error',
  DUPLICATE_CYCLE_NAME: 'This cycle name is already in use',
  DUPLICATE_PROCESS_NAME: 'This process name is already in use',
});

export const QUERY_STRING_PARAM = Object.freeze({
  KEY: {
    SERVICE: 'service',
    QUERY: 'query',
  },
  VALUE: {
    FALSE: 'false',
  },
});
