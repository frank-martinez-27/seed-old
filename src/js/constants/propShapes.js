import PropTypes from 'prop-types';

export const FetchedDataInitialState = {
  isFetching: false,
  error: {
    status: false,
    payload: {
      response: {},
    },
  },
};

export const FetchedDataArrayInitialState = {
  ...FetchedDataInitialState,
  payload: [],
};

export const FetchedDataObjectInitialState = {
  ...FetchedDataInitialState,
  payload: {},
};

export const FetchedDataShape = {
  isFetching: PropTypes.bool,
  paginationAction: PropTypes.string,
  payload: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  error: PropTypes.object,
};

export const fetchedDataError = action => ({
  status: true,
  payload: action.error,
});
