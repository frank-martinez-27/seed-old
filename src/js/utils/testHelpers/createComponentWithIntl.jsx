// https://github.com/yahoo/react-intl/blob/master/examples/jest-snapshot-testing/src/utils/createComponentWithIntl.js

import React from 'react';
import renderer from 'react-test-renderer'; // eslint-disable-line
import { IntlProvider } from 'react-intl';
import content from '../../constants/content.json';
import { getMessagesForLanguage } from '../../containers/Localization';
import { LANGUAGES } from '../../constants';

const createComponentWithIntl = (children, props = { locale: LANGUAGES.ENGLISH, messages: {} }) => {
  const translation = getMessagesForLanguage(props.locale, content.messages);
  return renderer.create(
    <IntlProvider {...props} messages={translation}>
      {children}
    </IntlProvider>
  );
};

export default createComponentWithIntl;
