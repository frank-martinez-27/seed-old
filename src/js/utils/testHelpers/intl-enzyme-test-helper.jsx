/**
 * https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#helper-function-1
 *
 * Components using the react-intl module require access to the intl context.
 * This is not available when mounting single components in Enzyme.
 * These helper functions aim to address that and wrap a valid,
 * English-locale intl context around them.
 */

import React from 'react';
import { IntlProvider, intlShape } from 'react-intl';
import { mount, shallow } from 'enzyme'; // eslint-disable-line import/no-extraneous-dependencies
import { LANGUAGES } from '../../constants';

const content = require('../../constants/content.json');

// Create the IntlProvider to retrieve context for wrapping around.
const intlProvider = new IntlProvider(
  { locale: LANGUAGES.ENGLISH, messages: content.messages.en },
  {}
);
const { intl } = intlProvider.getChildContext();

/**
 * When using React-Intl `injectIntl` on components, props.intl is required.
 */
function nodeWithIntlProp(node) {
  return React.cloneElement(node, { intl });
}

export function shallowWithIntl(node, { context } = {}) {
  return shallow(nodeWithIntlProp(node), {
    context: Object.assign({}, context, { intl }),
  });
}

export function mountWithIntl(node, { context, childContextTypes } = {}) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, { intl }),
    childContextTypes: Object.assign({}, { intl: intlShape }, childContextTypes),
  });
}

/*
 * Repeatedly render a component tree using enzyme.shallow() until
 * finding and rendering TargetComponent.
 *
 * The `componentInstance` parameter is a React component instance.
 * Example: <MyComponent {...props} />
 *
 * The `TargetComponent` parameter is the React class (or function) that
 * you want to retrieve from the component tree.
 */
export function shallowUntilTarget(
  componentInstance,
  TargetComponent,
  { maxTries = 10, shallowOptions, _shallow = shallow } = {}
) {
  if (!componentInstance) {
    throw new Error('componentInstance parameter is required');
  }
  if (!TargetComponent) {
    throw new Error('TargetComponent parameter is required');
  }

  let root = _shallow(componentInstance, shallowOptions);

  if (typeof root.type() === 'string') {
    // If type() is a string then it's a DOM Node.
    // If it were wrapped, it would be a React component.
    throw new Error('Cannot unwrap this component because it is not wrapped');
  }

  for (let tries = 1; tries <= maxTries; tries++) {
    if (root.is(TargetComponent)) {
      // Now that we found the target component, render it.
      return root.shallow(shallowOptions);
    }
    // Unwrap the next component in the hierarchy.
    root = root.dive();
  }

  throw new Error(
    `Could not find ${TargetComponent} in rendered instance: ${componentInstance}; gave up after ${maxTries} tries`
  );
}
