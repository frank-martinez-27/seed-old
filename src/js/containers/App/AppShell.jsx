import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ModalConductor from '../Modal/ModalConductor';
import { FetchedDataShape } from '../../constants/propShapes';
import { ToastContainer } from 'react-toastify';
import NavigationBar from '../../components/NavigationBar';

export function AppShell({ children, content }) {
  return (
    <div className="app" style={{ flexGrow: 1 }}>
      <NavigationBar />
      <ToastContainer />
      <main id="main-content" className="app__main-content" data-test-main-content>
        {children}
      </main>
      <ModalConductor />
    </div>
  );
}

AppShell.propTypes = {
  children: PropTypes.element.isRequired,
  content: PropTypes.shape(FetchedDataShape).isRequired,
};

const mapStateToProps = state => {
  const { content } = state;
  return {
    content,
  };
};

export default connect(mapStateToProps)(AppShell);
