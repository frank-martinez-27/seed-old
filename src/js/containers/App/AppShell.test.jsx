import React from 'react';
import { shallow } from 'enzyme';
import AppShell from '.';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store'; // eslint-disable-line import/no-extraneous-dependencies
import thunk from 'redux-thunk';
import createComponentWithIntl from '../../utils/testHelpers/createComponentWithIntl';
import { MODALS } from '../../constants';
import { IntlProvider } from 'react-intl';
import { mountWithIntl } from '../../utils/testHelpers/intl-enzyme-test-helper';
jest.mock('react-router-dom');

const APP_SHELL_DEFAULT_PROPS = {
  content: {
    payload: {},
  },
  currentModal: {
    modal: MODALS.NONE,
  },
};

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({
  ...APP_SHELL_DEFAULT_PROPS,
});

const APP_SHELL_COMPONENT = (
  <Provider store={store}>
    <IntlProvider locale="EN">
      <AppShell {...APP_SHELL_DEFAULT_PROPS}>
        <p>this content should render</p>
      </AppShell>
    </IntlProvider>
  </Provider>
);

describe('App Shell', () => {
  it('renders correctly with content', () => {
    const wrapper = createComponentWithIntl(APP_SHELL_COMPONENT).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    shallow(APP_SHELL_COMPONENT);
  });

  it('e2e-like test - renders welcome message', () => {
    const wrapper = mountWithIntl(APP_SHELL_COMPONENT);
    expect(wrapper.find('[data-test-main-content]')).toHaveLength(1);
    expect(wrapper.find('[data-test-main-content]').exists()).toEqual(true);
  });
});
