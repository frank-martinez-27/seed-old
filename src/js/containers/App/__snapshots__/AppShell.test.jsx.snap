// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Shell renders correctly with content 1`] = `
<div
  className="app"
  style={
    Object {
      "flexGrow": 1,
    }
  }
>
  <header
    className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorPrimary-7"
  >
    <div
      className="MuiToolbar-root-36 MuiToolbar-regular-38 MuiToolbar-gutters-37"
    >
      <button
        aria-label="Menu"
        className="MuiButtonBase-root-46 MuiIconButton-root-40 MuiIconButton-colorInherit-41"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "marginLeft": -12,
            "marginRight": 20,
          }
        }
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-45"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-49"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-56"
        />
      </button>
      <h2
        className="MuiTypography-root-63 MuiTypography-title-69 MuiTypography-colorInherit-82"
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <span>
          Welcome
        </span>
      </h2>
      <button
        className="MuiButtonBase-root-46 MuiButton-root-88 MuiButton-text-90 MuiButton-flat-93 MuiButton-colorInherit-109"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-89"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root-56"
        />
      </button>
    </div>
  </header>
  <div
    className="Toastify"
  />
  <main
    className="app__main-content"
    data-test-main-content={true}
    id="main-content"
  />
</div>
`;
